lms/djangoapps/commerce/api/v0/views.py

```diff
@@ -194,6 +192,17 @@ def post(self, request, *args, **kwargs):
         self._handle_marketing_opt_in(request, course_key, user)
         return response
 
+    def _add_request_cookie_to_api_session(self, server_session, request, cookie_name):
+        user_cookie = None
+        if cookie_name:
+            user_cookie = request.COOKIES.get(cookie_name)
+        if user_cookie:
+            server_cookie = {cookie_name: user_cookie}
+            if server_session.cookies:
```

Comment:
this is fine but an alternative way to do this could be something like

server_cookie = requests.utils.cookiejar_from_dict({cookie_name: user_cookie})
if not server_session.cookies:
  server_session.cookies = new cookie jar
server_session.cookies.update(server_cookie)